config/base/src/lib.rs

```diff
@@ -296,28 +433,30 @@ pub mod proxy {
 
     /// Trait for configuration loading and deserialization from
     /// the environment
-    pub trait LoadFromEnv {
-        /// Error type returned by methods of this trait
-        type Error;
+    pub trait LoadFromEnv: Sized {
+        /// The return type. Could be target `Configuration`,
+        /// some `Result`, `Option` as users see fit.
```

Comment:
```suggestion
        /// The return type. Could be target `Configuration`,
        /// `Result`, or `Option`.
```

Reply:
How about
```suggestion
        /// The return type. Could be target `Configuration`,
        /// some `Result`, `Option`, or any other reasonable alternative.
```
Just to signify that these three are the most likely options, but it could be anything else if desired.

Reply:
can you define `reasonable alternative`? what's reasonable and what is not?